### PR TITLE
Fixed #5500: implemented "Toggle Minimap" command

### LINKED ISSUE #5500: [monaco] missing "toggle minimap" command
We should add the "Toggle Minimap" command to the top bar menu and the quick-open.

### PR DESCRIPTION
**Pull request by fangnx**
- Implemented the "Toggle Minimap" command in the command pallete.

### CODE REVIEW

File: packages/editor/src/browser/editor-command.ts [hunk 1]
```diff
@@ -100,6 +100,14 @@ export namespace EditorCommands {
         category: 'View',
         label: 'Show All Opened Editors'
     };
+    /**
+     * Command that toggles the minimap.
+     */
+    export const TOGGLE_MINIMAP: Command = {
+        id: 'editor.minimap.enabled',
```
Comment (vince-fugnitto) [inline packages/editor/src/browser/editor-command.ts]: We should use a different id, this one is already being used by the preference and it becomes confusing.
https://github.com/theia-ide/theia/blob/d2d4f75897fdb13bcae20409ceef490ea225086e/packages/editor/src/browser/editor-preferences.ts#L129-L132

You should use the same as vscode: https://github.com/microsoft/vscode/blob/7bc29bf67c17608b11e23a8497c8f1de5a0922cf/src/vs/workbench/contrib/codeEditor/browser/toggleMinimap.ts#L15

File: packages/editor/src/browser/editor-navigation-contribution.ts [hunk 2]
```diff
@@ -88,6 +96,14 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
         this.toDispose.dispose();
     }
 
+    /**
+     * Toggle the display of minimap in the editor.
+     */
+    protected async doToggleMinimap(): Promise<void> {
+        const value: boolean | undefined = this.preferenceService.get('editor.minimap.enabled');
+        this.preferenceService.set('editor.minimap.enabled', !value);
```
Comment (vince-fugnitto) [inline packages/editor/src/browser/editor-navigation-contribution.ts]: It might be better to be explicit and state the scope: `PreferenceScope.User`.

Comment (fangnx) [reply packages/editor/src/browser/editor-navigation-contribution.ts]: Thanks! Just fixed.

File: packages/editor/src/browser/editor-command.ts (hunk 1 above)
Comment (fangnx) [reply packages/editor/src/browser/editor-command.ts]: Just changed it.

**Comment by vince-fugnitto** [comment]
@lmcbout can you try it out as well? Note that the command only updates the `User` preference.
If the preference already exists at the workspace or folder level it takes precedence over the toggle.

**Comment by lmcbout** [comment]
> @lmcbout can you try it out as well? Note that the command only updates the `User` preference.
> If the preference already exists at the workspace or folder level it takes precedence over the toggle.

1- Code should be modify  from doToggleMinimap() to toggleMinimap() as suggested
2- When the 'editor.minimap.enabled' is defined in the workspace, it takes over and nothing happen. If you want to have the same behavior as VSCode,  VSCode show/hide the minimap even when it is defined at the workspace level, so considerating this, it does not work. Do we want to show/hide the minimap if this preferences is defined at a lower level ?

**Comment by vince-fugnitto** [comment]
> 2- When the 'editor.minimap.enabled' is defined in the workspace, it takes over and nothing happen. If you want to have the same behavior as VSCode, VSCode show/hide the minimap even when it is defined at the workspace level, so considerating this, it does not work. Do we want to show/hide the minimap if this preferences is defined at a lower level ?

@lmcbout 

We have the same behaviour as VSCode.

VSCode sets the `User` preference only for the `editor.minimap.enabled` preference.
If the preference is set at the `Workspace` level it takes precedence over the `User`.

I do not believe there is anything more to do in this PR, and this comment was already discussed https://github.com/theia-ide/theia/pull/5633#issuecomment-508130982

And looking at their source code you can see as well we implement it similarly:
https://github.com/microsoft/vscode/blob/7bc29bf67c17608b11e23a8497c8f1de5a0922cf/src/vs/workbench/contrib/codeEditor/browser/toggleMinimap.ts#L26-L29

**Comment by fangnx** [comment]
> Would be nice, but not mandatory: doToggleMinimap() to toggleMinimap() as suggested
> After further investigation, VSCode behave the same. As long as you don't put the preference in the workspace level, it is fine. VSCode put as default "true" in the workspace .
> LGTM

I will change the naming ;). Thanks for the feedback!

**Pull request merged**